o2a/transformers/base_transformer.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base transformer classes"""
+from abc import ABC, abstractmethod
+from typing import Type
+
+from o2a.converter.parsed_action_node import ParsedActionNode
+from o2a.converter.workflow import Workflow
+
+
+# pylint: disable=too-few-public-methods
+class BaseWorkflowTransformer(ABC):
+    """
+    Base class for all transformers
+    """
+
+    @abstractmethod
+    def process_workflow(self, workflow: Workflow):
+        pass
+
+
+class TypeNodeWorkflowTransformer(BaseWorkflowTransformer):
+    """
+    A transformer that lets you simplify the processing of one type of node.
+    """
+
+    @property
+    @abstractmethod
+    def wanted_type(self) -> Type:
```

Comment:
I would rather name it as "transformed_types". The "wanted_type" does not really explain what it is about.
And I think it should be -> Iterable[Type] as return value. I can easily imagine transformer which applies to several mappers.

Reply:
And for example if you want to do what you did in "inaccessible nodes" -> this method should return [] list which means "all types". This will simplify the class hierarchy.

Reply:
We have a very small list of types. In imagination I am able to imagine case when it could be useful, but looking at the list of mapper is difficult. I think it is not necessary, but it will change it.

Reply:
I removed this classes.